tests/test_custom_expert.py

```diff
@@ -0,0 +1,55 @@
+import os
+import pytest
+import torch
+import hivemind
+
+from typing import Optional
+from hivemind import RemoteExpert, background_server
+
+@pytest.mark.forked
+def test_custom_expert(port: Optional[int] = None, hid_dim=16):
+    with background_server(
+        expert_cls='perceptron', num_experts=2, device='cpu',
+        hidden_dim=hid_dim, num_handlers=2, no_dht=True,
+        custom_module_path=os.path.join(os.path.dirname(__file__), 'custom_networks.py')) as (server_endpoint, _):
+
+        expert0 = RemoteExpert('expert.0', server_endpoint)
+        expert1 = RemoteExpert('expert.1', server_endpoint)
+
+        for step in range(1, 4):
+            batch = torch.randn(step, hid_dim)
+
+            output0 = expert0(batch)
+            output1 = expert1(batch)
+
+            loss = output0.sum()
+            loss.backward()
+            loss = output1.sum()
+            loss.backward()
+
+@pytest.mark.forked
+def test_multihead_expert(port: Optional[int] = None, hid_dim=16):
+    with background_server(
+        expert_cls='multihead', num_experts=2, device='cpu',
+        hidden_dim=hid_dim, num_handlers=2, no_dht=True,
+        custom_module_path=os.path.join(os.path.dirname(__file__), 'custom_networks.py')) as (server_endpoint, _):
+
+        expert0 = RemoteExpert('expert.0', server_endpoint)
+        expert1 = RemoteExpert('expert.1', server_endpoint)
+
+        for step in range(1, 4):
+            batch = (torch.randn(step, hid_dim), torch.randn(step, 2 * hid_dim), torch.randn(step, 3 * hid_dim))
+
+            output0 = expert0(*batch)
+            output1 = expert1(*batch)
+
+            loss = output0.sum()
+            loss.backward()
+            loss = output1.sum()
+            loss.backward()
```

Comment:
Is this test necessary in presence of `test_custom_expert` or vice versa?

Reply:
If i remember the counterargument correctly:
- this test verifies that custom experts support multiple inputs
- this test verifies that custom expert inputs handle higher ndims
- this test verifies that the new code can handle multiple custom experts correctly

Reply:
I was checking that custom expert works correctly with inputs consisting of multiple tensors e.g. transformer's layers which take input and mask